affectedmoduledetector/src/main/kotlin/com/dropbox/affectedmoduledetector/AffectedModuleDetectorPlugin.kt

```diff
@@ -77,10 +77,7 @@ class AffectedModuleDetectorPlugin : Plugin<Project> {
     }
 
     private fun registerCustomTasks(rootProject: Project) {
-        val mainConfiguration = requireNotNull(
-            value = rootProject.extensions.findByName(AffectedModuleConfiguration.name),
-            lazyMessage = {  "Unable to find ${AffectedTestConfiguration.name} in $rootProject" }
-        ) as AffectedModuleConfiguration
+        val mainConfiguration = requireConfiguration(rootProject)
```

Comment:
this method calls quite often in big multymodule projects. 
1) too much boilerplate 
2) [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself#:~:text=The%20DRY%20principle%20is%20stated,their%20book%20The%20Pragmatic%20Programmer.)
3) we can do  small optimization with singleton

Reply:
I recognized your comments in my previous PR. Actually, I was wrong. Thank u for the explanation.  

I closed my previous PR, a bit pity my time. But it is the right way. 

We can merge your version for solving the problem.